lib/rules/space-unary-ops.js

```diff
@@ -65,7 +65,8 @@ module.exports = function(context) {
             firstToken = tokens[0],
             secondToken = tokens[1];
 
-        if ((node.type === "NewExpression" || node.prefix) && firstToken.type === "Keyword") {
+
+        if (((node.type === "NewExpression" || node.prefix) && firstToken.type === "Keyword") || (firstToken.type === "Identifier" && firstToken.value === "yield")) {
```

Comment:
I think there is a bug in espree, the token for the `yield` keyword shouldn’t be `Identifier`. I’ve filed an espree issue eslint/espree#190